js/listado.js

```diff
@@ -38,9 +38,8 @@ Listado.prototype.obtenerUbicaciones = function() {
     //Array donde se van a ir agregando las ciudades (van a estar repetidas)
     var ciudadesRep = [];
     //Se recorre el array de restaurantes y se va agregando al array creado, todas las ubicaciones o ciudades encontradas
-    for (var i = 0; i < this.restaurantes.length; i++) {
-        ciudadesRep.push(this.restaurantes[i].ubicacion);
-    }
+
+    this.restaurantes.map(res => ciudadesRep.push(res.ubicacion))
```

Comment:
loco, acordate que `map` al igual que `filter` y `reduce` te devuelven un array nuevo. Esto que hiciste creo que deberia funcionar, pero se ve raro. Vos por lo general queres asignar a alguna variable (o retornar) el resultado de una llamada a `map`.
```js
const ciudadesRep = this.restaurantes.map(res => res.ubicacion)
```

Reply:
La razon por la cual se ve raro es que usaste `map` como si fuera un `for`. Por cada elemento del array terminas haciendo un push. Pero el map en realidad lo tenes que pensar como una transformacion que se hace sobre cada elemento del array. En el codigo de ejemplo que te pase la transformacion es "cada restaurante lo transformo en la ubicacion del restaurante".